google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2501,6 +2540,44 @@ private void removeLongRunningSessions(
         }
       }
     }
+
+    void maintainMultiplexedSession(Instant currentTime) {
+      try {
+        if (options.getUseMultiplexedSession()) {
+          synchronized (lock) {
+            if (getMultiplexedSession().get() != null && isMultiplexedSessionStale(currentTime)) {
```

Comment:
nit: add a check for `getMultiplexedSession().is done()` at the start of the if statement. This prevents us from blocking here in the theoretical case that the maintainer tries to maintain the multiplexed session before it has been created. We should never call any blocking methods when holding the `lock`.